### PR TITLE
[ElectedRepresentative][ADMIN] Add UserListDefinition and its filter

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

**Comment by illusionOfParadise** [comment]
[ch11497]

**Comment by illusionOfParadise** [comment]
[ch11498]

File: src/Entity/ElectedRepresentative/ElectedRepresentative.php [hunk 1]
```diff
@@ -118,16 +118,23 @@ class ElectedRepresentative
...
+    private $isSupportingLaREM = false;
 
     /**
      * @var bool|null
      *
-     * @ORM\Column(type="boolean", nullable=true)
+     * @ORM\Column(type="boolean", options={"default": false})
      */
-    private $hasFollowedTraining;
+    private $isInstancesMember = false;
```
Comment (kevin-verschaeve) [inline src/Entity/ElectedRepresentative/ElectedRepresentative.php:130]: can we rename it `$instanceMember`, so the getter become `isInstanceMember()` ? (If so, don't forget to update the migration, or the column name here)

Comment (illusionOfParadise) [reply src/Entity/ElectedRepresentative/ElectedRepresentative.php]: There is no more `instanceMember`, the structure has been completely changed.

**Comment by illusionOfParadise** [comment]
[ch11514]

File: src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php [hunk 2]
```diff
@@ -451,6 +449,16 @@ protected function configureDatagridFilters(DatagridMapper $datagridMapper)
                     }
                 },
             ])
+            ->add('userListDefinitions', ModelAutocompleteFilter::class, [
```
Comment (ottaviano) [inline src/Admin/ElectedRepresentative/ElectedRepresentativeAdmin.php:452]: do we need to check here the specific UserList type for avoiding to display referent user lists?

File: src/Entity/ElectedRepresentative/ElectedRepresentative.php [hunk 3]
```diff
@@ -23,6 +24,7 @@
  */
 class ElectedRepresentative
 {
+    use EntityUserListDefinitionTrait;
```
Comment (ottaviano) [inline src/Entity/ElectedRepresentative/ElectedRepresentative.php:27]: an empty line is missing after `use`

File: src/Entity/UserListDefinition.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        $this->type = $type;
+        $this->label = $label;
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getType(): ?string
```
Comment (ottaviano) [inline src/Entity/UserListDefinition.php:58]: `: string`

File: src/Entity/UserListDefinition.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        return $this->type;
+    }
+
+    public function setType(string $type): void
+    {
+        $this->type = $type;
+    }
+
+    public function getLabel(): ?string
```
Comment (ottaviano) [inline src/Entity/UserListDefinition.php:68]: `: string`

File: src/Entity/UserListDefinition.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public function setLabel(string $label): void
+    {
+        $this->label = $label;
+    }
+
+    public function __toString(): string
+    {
+        return (string) $this->label;
```
Comment (ottaviano) [inline src/Entity/UserListDefinition.php:80]: `return $this->label;`
as $label is required construct string argument

File: src/Entity/UserListDefinitionEnum.php [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace App\Entity;
+
+use MyCLabs\Enum\Enum;
+
+final class UserListDefinitionEnum extends Enum
+{
+    public const TYPE_ELECTED_REPRESENTATIVE = 'elected_representative';
+
+    public const LABEL_ELECTED_REPRESENTATIVE_SUPPORTING_LA_REM = 'Sympathisant(e) LaREM';
```
Comment (ottaviano) [inline src/Entity/UserListDefinitionEnum.php:11]: I think we don't need to have a const for the label as each label will be managed by admin

**Pull request merged**